llvm/test/CodeGen/AArch64/ext-narrow-index.ll

```diff
@@ -118,11 +117,16 @@ entry:
 }
 
 define <4 x i16> @i16_off8(<8 x i16> %arg1, <8 x i16> %arg2) {
-; CHECK-LABEL: i16_off8:
-; CHECK:       // %bb.0: // %entry
-; CHECK-NEXT:    mov v0.16b, v1.16b
-; CHECK-NEXT:    // kill: def $d0 killed $d0 killed $q0
-; CHECK-NEXT:    ret
+; CHECK-SD-LABEL: i16_off8:
+; CHECK-SD:       // %bb.0: // %entry
+; CHECK-SD-NEXT:    mov v0.16b, v1.16b
+; CHECK-SD-NEXT:    // kill: def $d0 killed $d0 killed $q0
+; CHECK-SD-NEXT:    ret
+;
+; CHECK-GISEL-LABEL: i16_off8:
+; CHECK-GISEL:       // %bb.0: // %entry
+; CHECK-GISEL-NEXT:    // kill: def $d0 killed $d0 killed $q0
```

Comment:
This exposes a bug:
```
Erasing: %2:_(<4 x s16>) = G_SHUFFLE_VECTOR %0:_(<8 x s16>), %1:_, shufflemask(8, 9, 10, 11)

Created: %2:_(<4 x s16>), %3:_(<4 x s16>) = G_UNMERGE_VALUES %0:_(<8 x s16>)
```

The shufflemask selects the 4 lower elements of `%1`, while the `G_UNMERGE_VALUES` gets them from `%0`.

This is the incoming MIR:
```
bb.1.entry:
  liveins: $q0, $q1
  %0:_(<8 x s16>) = COPY $q0
  %1:_(<8 x s16>) = COPY $q1
  %2:_(<4 x s16>) = G_SHUFFLE_VECTOR %0:_(<8 x s16>), %1:_, shufflemask(8, 9, 10, 11)
  $d0 = COPY %2:_(<4 x s16>)
  RET_ReallyLR implicit $d0
```

Reply:
This was resolved in the latest push